models/aeif_cond_beta_multisynapse.cpp

```diff
@@ -168,60 +156,54 @@ aeif_cond_beta_multisynapse::Parameters_::set( const DictionaryDatum& d )
   updateValue< double >( d, names::C_m, C_m );
   updateValue< double >( d, names::g_L, g_L );
 
-  std::vector< double > Erev_tmp, taur_tmp, taud_tmp;
+  size_t old_n_receptors = n_receptors();
   bool Erev_flag =
-    updateValue< std::vector< double > >( d, names::E_rev, Erev_tmp );
+    updateValue< std::vector< double > >( d, names::E_rev, E_rev );
   bool taur_flag =
-    updateValue< std::vector< double > >( d, names::taus_rise, taur_tmp );
+    updateValue< std::vector< double > >( d, names::taus_rise, taus_rise );
   bool taud_flag =
-    updateValue< std::vector< double > >( d, names::taus_decay, taud_tmp );
+    updateValue< std::vector< double > >( d, names::taus_decay, taus_decay );
   if ( Erev_flag || taur_flag || taud_flag )
```

Comment:
I agree that the code can be made more compact if we simply want to check consistency. However my idea is that we should not only check for consistency, but also give the user a feedback that suggests him how he should modify his command. For instance, if the current number of receptor ports is 4 (or if they have not yet been initialized so the number is 0 or 1) and the user types a command as:
nest.SetStatus(neuron, {'taus_decay':[50.0,20.0,20.0]})
I think a message as:
"If the number of receptor ports is changed, all three arrays E_rev, taus_rise and taus_decay must be provided."
is more helpful to the user than:
"E_rev, taus_rise and taus_decay must have the same size."
because with the latter message it may not be obvious to all users that the problem can be solved by providing all three arrays as arguments.